_docs/tasks/traffic-management-v1alpha3/ingress.md

```diff
@@ -0,0 +1,579 @@
+   we could add the special value `mesh` to the list of `gateways`.
+        
+### Verifying a Gateway
+   
+The proxy instances implementing a particular `Gateway` configuration can be specified using a
+[selector]({{home}}/docs/reference/config/istio.networking.v1alpha3.html#Gateway.selector) field.
+If not specified, as in our case, the `Gateway` will be implemented by the default `istio-ingress` controller.
+Therefore, to test our Gateway we will send requests to the `istio-ingress` service.
+
+1. Get the ingress controller pod's hostIP:
+
+   ```bash
+   kubectl -n istio-system get po -l istio=ingress -o jsonpath='{.items[0].status.hostIP}'
+   ```
+
+   ```bash
+   169.47.243.100
+   ```
+
+1. Get the istio-ingress service's nodePorts for port 80 and 443:
+
+   ```bash
+   kubectl -n istio-system get svc istio-ingress
+   ```
+
+   ```bash
+   NAME            CLUSTER-IP     EXTERNAL-IP   PORT(S)                      AGE
+   istio-ingress   10.10.10.155   <pending>     80:31486/TCP,443:32254/TCP   32m
+   ```
+
+   ```bash
+   export INGRESS_HOST=169.47.243.100:31486
+   export SECURE_INGRESS_HOST=169.47.243.100:32254
+   ```
+
+1. Access the httpbin service with either HTTP or HTTPS using _curl_:
+
+   ```bash
+   curl -I http://$INGRESS_HOST/status/200
+   ```
+
+   ```
+   HTTP/1.1 200 OK
+   server: envoy
+   date: Mon, 29 Jan 2018 04:45:49 GMT
+   content-type: text/html; charset=utf-8
+   access-control-allow-origin: *
+   access-control-allow-credentials: true
+   content-length: 0
+   x-envoy-upstream-service-time: 48
+   ```
+
+   ```bash
+   curl -I -k https://$SECURE_INGRESS_HOST/status/200
+   ```
+
+   ```
+   HTTP/1.1 200 OK
+   server: envoy
+   date: Mon, 29 Jan 2018 04:45:49 GMT
+   content-type: text/html; charset=utf-8
+   access-control-allow-origin: *
+   access-control-allow-credentials: true
+   content-length: 0
+   x-envoy-upstream-service-time: 96
+   ```
+
+1. Access any other URL that has not been explicitly exposed. You should
+   see a HTTP 404 error:
+
+   ```bash
+   curl -I http://$INGRESS_HOST/headers
+   ```
+
+   ```
+   HTTP/1.1 404 Not Found
+   date: Mon, 29 Jan 2018 04:45:49 GMT
+   server: envoy
+   content-length: 0
+   ```
+
+   ```bash
+   curl -I https://$SECURE_INGRESS_HOST/headers
+   ```
+
+   ```
+   HTTP/1.1 404 Not Found
+   date: Mon, 29 Jan 2018 04:45:49 GMT
+   server: envoy
+   content-length: 0
+   ```
+
+## Configuring ingress using a Kubernetes Ingress resource
+
+An Istio Ingress specification is based on the standard [Kubernetes Ingress Resource](https://kubernetes.io/docs/concepts/services-networking/ingress/)
+specification, with the following differences:
+
+1. Istio Ingress specification contains a `kubernetes.io/ingress.class: istio` annotation.
+
+2. All other annotations are ignored.
+
+The following are known limitations of Istio Ingress:
+
+1. Regular expressions in paths are not supported.
```

Comment:
We can tackle this in a better way by setting all path matches to be regex matches.. So, instead, we should tell people that the Istio regex format is c++11 regex format (http://en.cppreference.com/w/cpp/regex/ecmascript).

The main limitation of ingress is that virtual services cannot be combined with Ingress specs. Essentially, whatever is in ingress will just work as any normal ingress (path/prefix match to one backend service). DestinationRules will work as they apply to the service. However, virtual services (Traffic splitting, mirroring, header match, etc.) will not work, nor will fault injection. 

In other words, the ingress user will get the same feature that the ingress spec supports. And nothing more than that.